pkg/graphite/convert/whisperconverter/whisper_test.go

```diff
@@ -127,6 +128,176 @@ func TestExtractWhisperPoints(t *testing.T) {
+					// 1030     1020     1009     994
+					//  [XXXXXXXXX]
+					//  [          XXXXXXXXX]
+					//  [                   XXXXXXXX]
+
+					simpleArchiveInfo(11, 1),
+					simpleArchiveInfo(8, 3),
+					simpleArchiveInfo(7, 6),
+				},
+				points: [][]whisper.Point{
+					{
+						whisper.NewPoint(time.Unix(1020, 0), 20),
+						whisper.NewPoint(time.Unix(1021, 0), 21),
+						whisper.NewPoint(time.Unix(1022, 0), 22),
+						whisper.NewPoint(time.Unix(1023, 0), 23),
+						whisper.NewPoint(time.Unix(1024, 0), 24),
+						whisper.NewPoint(time.Unix(1025, 0), 25),
+						whisper.NewPoint(time.Unix(1026, 0), 26),
+						whisper.NewPoint(time.Unix(1027, 0), 27),
+						whisper.NewPoint(time.Unix(1028, 0), 28),
+						whisper.NewPoint(time.Unix(1029, 0), 29),
+						whisper.NewPoint(time.Unix(1030, 0), 30),
+					},
+					{
+						whisper.NewPoint(time.Unix(1009, 0), 9), // This is the lower border of this archive, this point should be kept
+						whisper.NewPoint(time.Unix(1012, 0), 12),
+						whisper.NewPoint(time.Unix(1015, 0), 15),
+						whisper.NewPoint(time.Unix(1018, 0), 18),
+						whisper.NewPoint(time.Unix(1021, 0), 21), // skipped
+						whisper.NewPoint(time.Unix(1024, 0), 24), // skipped
+						whisper.NewPoint(time.Unix(1027, 0), 27), // skipped
+						whisper.NewPoint(time.Unix(1030, 0), 30), // skipped
+					},
+					{
+						whisper.NewPoint(time.Unix(1000, 0), 0),
+						whisper.NewPoint(time.Unix(1006, 0), 6),
+						whisper.NewPoint(time.Unix(1009, 0), 99), // skipped because archive 1 has a point at this time
+						whisper.NewPoint(time.Unix(1012, 0), 12), // skipped
+						whisper.NewPoint(time.Unix(1018, 0), 18), // skipped
+						whisper.NewPoint(time.Unix(1024, 0), 24), // skipped
+						whisper.NewPoint(time.Unix(1030, 0), 30), // skipped
+					},
+				},
+			},
+			want: []whisper.Point{
+				{
+					Timestamp: 1000,
+					Value:     0,
+				},
+				{
+					Timestamp: 1006,
+					Value:     6,
+				},
+				{
+					Timestamp: 1009,
+					Value:     9,
+				},
+				{
+					Timestamp: 1012,
+					Value:     12,
+				},
+				{
+					Timestamp: 1015,
+					Value:     15,
+				},
+				{
+					Timestamp: 1018,
+					Value:     18,
+				},
+				{
+					Timestamp: 1020,
+					Value:     20,
+				},
+				{
+					Timestamp: 1021,
+					Value:     21,
+				},
+				{
+					Timestamp: 1022,
+					Value:     22,
+				},
+				{
+					Timestamp: 1023,
+					Value:     23,
+				},
+				{
+					Timestamp: 1024,
+					Value:     24,
+				},
+				{
+					Timestamp: 1025,
+					Value:     25,
+				},
+				{
+					Timestamp: 1026,
+					Value:     26,
+				},
+				{
+					Timestamp: 1027,
+					Value:     27,
+				},
+				{
+					Timestamp: 1028,
+					Value:     28,
+				},
+				{
+					Timestamp: 1029,
+					Value:     29,
+				},
+				{
+					Timestamp: 1030,
+					Value:     30,
+				},
+			},
+		},
+		{
+			name:       "test retention when first archives are empty",
+			metricName: "mymetric",
```

Comment:
This seems to be unrealistic scenario to me. If first archive (with raw data) is empty, how can there be any aggregations at all?

Reply:
an end-to-end test broke, so we added this test to cover that edge case

Reply:
:+1: